lib/rules/comment-pattern/__tests__/index.js

```diff
@@ -28,11 +28,46 @@ testRule({
 	reject: [
 		{
 			code: '/* not foo- */',
-			message: messages.expected,
+			message: messages.expected(/foo-.+/),
 		},
 		{
 			code: '/**/',
-			message: messages.expected,
+			message: messages.expected(/foo-.+/),
+		},
+	],
+});
+
+testRule({
+	ruleName,
+	config: ['foo-.+'],
+
+	accept: [
+		{
+			code: '/* foo-valid yay */',
+		},
+		{
+			code: `/* foo--
+			multi-line
+			\n
+			\r
+			\r\n
+			\n\r
+			\t
+				*/`,
+		},
+		{
+			code: '.without-comment {}',
+		},
+	],
+
+	reject: [
+		{
+			code: '/* not foo- */',
+			message: messages.expected(/foo-.+/),
+		},
+		{
+			code: '/**/',
+			message: messages.expected(/foo-.+/),
```

Comment:
It should match config, and in config we have string.

```suggestion
			message: messages.expected('foo-.+'),
		},
		{
			code: '/**/',
			message: messages.expected('foo-.+'),
```

Reply:
@hudochenkov got it, fixed now. 
I thought that we wanted to show the `normalizedPattern` in the message, instead of the pattern that was supplied in the config. I've updated the implementation + tests as suggested :)

Reply:
> I thought that we wanted to show the normalizedPattern in the message, instead of the pattern that was supplied in the config.

I didn't thought about that :) Now I'm not sure which one is better :)

@stylelint/contributors what do you think?

Reply:
It seems reasonable to me to always show the same form `pattern "foo"`, not `pattern "/foo/"`.
Even in different config file formats, it seems users expect the same output of a message, for example:

```yaml
# .stylelintrc.yml
rules:
  comment-pattern: ["foo"]
```

```js
// .stylelintrc.js
module.exports = {
  rules: {
    "comment-pattern": [/foo/] // equal to ["foo"]
  }
}
```

```console
$ bin/stylelint.js a.css --config .stylelintrc.yml
a.css
 1:1  ✖  Expected comment to match specified pattern "foo"   comment-pattern

$ bin/stylelint.js a.css --config .stylelintrc.js
a.css
 1:1  ✖  Expected comment to match specified pattern "/foo/"   comment-pattern
```

Reply:
@ybiquitous forgive me but I might be confused.. Are you suggesting to strip the slashes from the pattern when generating an error message? If so, how should regexes with modifiers look? e.g. `/foo-.+/i` or `/foo-.+/ig`?

Reply:
> Are you suggesting to strip the slashes from the pattern when generating an error message?

Yes, I thought we could keep the same appearance of an error message in any config formats.

> If so, how should regexes with modifiers look? e.g. `/foo-.+/i` or `/foo-.+/ig`?

Sorry, I missed regex modifiers. Surely your current implementaion seems better if we need to include the modifiers. 👍